solidity/ecdsa/test/WalletRegistryGovernance.test.ts

```diff
@@ -0,0 +1,966 @@
+          .beginMaliciousDkgResultNotificationRewardMultiplierUpdate(100)
+
+        await helpers.time.increaseTime(11 * 60 * 60) // 11 hours
+
+        await expect(
+          walletRegistryGovernance
+            .connect(governance)
+            .finalizeMaliciousDkgResultNotificationRewardMultiplierUpdate()
+        ).to.be.revertedWith("Governance delay has not elapsed")
+
+        await restoreSnapshot()
+      })
+    })
+
+    context(
+      "when the update process is initialized and governance delay passed",
+      () => {
+        let tx
+
+        before(async () => {
+          await createSnapshot()
+
+          await walletRegistryGovernance
+            .connect(governance)
+            .beginMaliciousDkgResultNotificationRewardMultiplierUpdate(100)
+
+          await helpers.time.increaseTime(12 * 60 * 60) // 12 hours
+
+          tx = await walletRegistryGovernance
+            .connect(governance)
+            .finalizeMaliciousDkgResultNotificationRewardMultiplierUpdate()
+        })
+
+        after(async () => {
+          await restoreSnapshot()
+        })
+
+        it("should update the DKG malicious result notification reward multiplier", async () => {
+          expect(
+            await walletRegistry.maliciousDkgResultNotificationRewardMultiplier()
+          ).to.be.equal(100)
+        })
+
+        it("should emit MaliciousDkgResultNotificationRewardMultiplierUpdated event", async () => {
+          await expect(tx)
+            .to.emit(
+              walletRegistryGovernance,
+              "MaliciousDkgResultNotificationRewardMultiplierUpdated"
+            )
+            .withArgs(100)
+        })
+
+        it("should reset the governance delay timer", async () => {
+          await expect(
+            walletRegistryGovernance.getRemainingMaliciousDkgResultNotificationRewardMultiplierUpdateTime()
+          ).to.be.revertedWith("Change not initiated")
+        })
+      }
+    )
+  })
+
+  describe("beginDkgResultChallengePeriodLengthUpdate", () => {
+    context("when the caller is not the owner", () => {
+      it("should revert", async () => {
+        await expect(
+          walletRegistryGovernance
+            .connect(thirdParty)
+            .beginDkgResultChallengePeriodLengthUpdate(11)
+        ).to.be.revertedWith("Ownable: caller is not the owner")
+      })
+    })
+
+    context("when the update value is less than 10", () => {
+      it("should revert", async () => {
+        await expect(
+          walletRegistryGovernance
+            .connect(governance)
+            .beginDkgResultChallengePeriodLengthUpdate(9)
+        ).to.be.revertedWith("DKG result challenge period length must be >= 10")
+      })
+    })
+
+    context("when the update value is at least 10", () => {
+      it("should accept the value", async () => {
+        await createSnapshot()
+
+        await walletRegistryGovernance
+          .connect(governance)
+          .beginDkgResultChallengePeriodLengthUpdate(10)
```

Comment:
We could use `to.not.be.reverted` for verification (in other places too).

```suggestion
        await expect(
          walletRegistryGovernance
            .connect(governance)
            .beginDkgResultChallengePeriodLengthUpdate(10)
          ).to.not.be.reverted
```